src/api/eth.rs

```diff
@@ -72,9 +72,16 @@ impl<T: Transport> Eth<T> {
     /// Call a contract without changing the state of the blockchain to estimate gas usage.
     pub fn estimate_gas(&self, req: CallRequest, block: Option<BlockNumber>) -> CallFuture<U256, T::Out> {
         let req = helpers::serialize(&req);
-        let block = helpers::serialize(&block.unwrap_or(BlockNumber::Latest));
 
-        CallFuture::new(self.transport.execute("eth_estimateGas", vec![req, block]))
+        let args = {
```

Comment:
Minor style nitpick:
```rust
let args = match block {
  Some(block) => vec![req, helpers::serialize(&block)],
  None => vec![req],
};
```
would be my preferred style.

Reply:
Fixed